lfs/config.go

```diff
@@ -315,48 +312,85 @@ func (c *Configuration) loadGitConfig() bool {
 	configFile := filepath.Join(LocalWorkingDir, ".gitconfig")
 	fileOutput, err := git.Config.ListFromFile(configFile)
 	if err != nil {
-		panic(fmt.Errorf("Error listing git config from file: %s", err))
+		panic(fmt.Errorf("Error listing git config from %s: %s", configFile, err))
 	}
 
 	localConfig := filepath.Join(LocalGitDir, "config")
 	localOutput, err := git.Config.ListFromFile(localConfig)
 	if err != nil {
-		panic(fmt.Errorf("Error listing git config from file %s", err))
+		panic(fmt.Errorf("Error listing git config from %s: %s", localConfig, err))
 	}
 
-	output = fileOutput + "\n" + listOutput + "\n" + localOutput
+	uniqRemotes := make(map[string]bool)
+	c.readGitConfig(fileOutput, uniqRemotes, true)
+	c.readGitConfig(listOutput, uniqRemotes, false)
+	c.readGitConfig(localOutput, uniqRemotes, false)
+
+	c.remotes = make([]string, 0, len(uniqRemotes))
+	for remote, isOrigin := range uniqRemotes {
+		if isOrigin {
+			continue
+		}
+		c.remotes = append(c.remotes, remote)
+	}
 
+	return true
+}
+
+func (c *Configuration) readGitConfig(output string, uniqRemotes map[string]bool, whitelisted bool) {
 	lines := strings.Split(output, "\n")
 	for _, line := range lines {
 		pieces := strings.SplitN(line, "=", 2)
 		if len(pieces) < 2 {
 			continue
 		}
+
+		allowed := !whitelisted
 		key := strings.ToLower(pieces[0])
 		value := pieces[1]
-		c.gitConfig[key] = value
 
 		keyParts := strings.Split(key, ".")
-		if len(keyParts) > 1 && keyParts[0] == "remote" {
-			remote := keyParts[1]
-			uniqRemotes[remote] = remote == "origin"
-		} else if len(keyParts) == 4 && keyParts[0] == "lfs" && keyParts[1] == "extension" {
+		if len(keyParts) == 4 && keyParts[0] == "lfs" && keyParts[1] == "extension" {
 			name := keyParts[2]
 			ext := c.extensions[name]
 			switch keyParts[3] {
 			case "clean":
+				if whitelisted {
+					continue
+				}
 				ext.Clean = value
 			case "smudge":
+				if whitelisted {
+					continue
+				}
 				ext.Smudge = value
 			case "priority":
+				allowed = true
 				p, err := strconv.Atoi(value)
 				if err == nil && p >= 0 {
 					ext.Priority = p
 				}
 			}
+
 			ext.Name = name
 			c.extensions[name] = ext
-		} else if len(keyParts) == 2 && keyParts[0] == "lfs" && keyParts[1] == "fetchinclude" {
+		} else if len(keyParts) > 1 && keyParts[0] == "remote" {
+			if whitelisted && (len(keyParts) < 3 || keyParts[2] != "lfsurl") {
```

Comment:
It would break for any subkeys under `remote.{name}.lfsurl`. I don't anticipate this being an issue in this case though.